testing/mock_raster_cache.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef TESTING_MOCK_RASTER_CACHE_H_
+#define TESTING_MOCK_RASTER_CACHE_H_
+
+#include "flutter/flow/raster_cache.h"
+#include "third_party/skia/include/core/SkImage.h"
+#include "third_party/skia/include/core/SkPicture.h"
+
+namespace flutter {
+namespace testing {
+
+// A |RasterCacheResult| implementation that represents a cached |Layer| or
+// |SkPicture| without the overhead of storage. This implementation is used
+// by |MockRasterCacheImageDelegate| only for testing proper usage of the
+// |RasterCache| in layer unit tests.
+class MockRasterCacheResult : public RasterCacheResult {
```

Comment:
It seems to me that we only need to do the following so we'll only have a single `RasterCacheResult` class instead of two classes `RasterCacheResult`, `SkRasterCacheResult`:

```
class RasterCacheResult {
 public:
  RasterCacheResult() = default;

  RasterCacheResult(const RasterCacheResult& other) = default;

  RasterCacheResult(sk_sp<SkImage> image, const SkRect& logical_rect);

  virtual operator bool() const { return static_cast<bool>(image_); }

  virtual bool is_valid() const { return static_cast<bool>(image_); };

  virtual void draw(SkCanvas& canvas, const SkPaint* paint = nullptr) const;

  virtual SkISize image_dimensions() const {
    return image_ ? image_->dimensions() : SkISize::Make(0, 0);
  };

 protected:
  sk_sp<SkImage> image_;
  SkRect logical_rect_;
};

class MockRasterCacheResult : public RasterCacheResult {
  MockRasterCacheResult(SkIRect device_rect);

  void draw(SkCanvas& canvas, const SkPaint* paint = nullptr) const override{};

  SkISize image_dimensions() const override { return device_rect_.size(); };

  operator bool() const override { return static_cast<bool>(image_); }

  virtual bool is_valid() const override { return static_cast<bool>(image_); };

private:
  SkIRect device_rect_;  
}
```

Reply:
So basically just incur the vestigial cost of the unused SkRect and the SkImage pointer in the Mock class?

Reply:
Yes, who cares the performance of a Mock class :D

Reply:
OK, done...